### PR TITLE
Add Edge Support

### LINKED ISSUE #2562: Get into Edge Browser
Edge added some self-service extension-adding tools, so we should be able to apply now:

https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/packaging/running-the-windows-app-certification-kit

### PR DESCRIPTION
**Pull request by scsaba**
As [Edge does not support PBKDF2 algorithm](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/9259365/) at the moment, I had to use the [asmcrypto](https://github.com/asmcrypto/asmcrypto.js) package. It is used in case of Edge browser only.

### CODE REVIEW

**Comment by scsaba** [comment]
added tests

**Comment by scsaba** [comment]
Fixes #2562

**Comment by scsaba** [comment]
I finished working on this issue, please review my pull request.

**Comment by Zanibas** [comment]
We are currently in-process of putting it through our pipeline

**Review by danfinlay — APPROVED** [review]
This looks like a good, safe change. I just need to verify that it fixes the Edge compat issue before approving the bounty...

File: app/scripts/edge-encryptor.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+const asmcrypto = require('asmcrypto.js')
+const Unibabel = require('browserify-unibabel')
```
Comment (danfinlay) [inline app/scripts/edge-encryptor.js]: Needs to be added to dependencies.

**Comment by tmashuang** [comment]
First run through looks really good. The only issue is downloading files such as the JSON state logs and the initial account creation seed phrase text file.  Which are features helps users backup their accounts and the other is for users sending support logs.

**Review by danfinlay — CHANGES_REQUESTED** [review]
Thomas tried this out and said that address copying still wasn't working, I'd like a verification that this works before approving.

**Comment by scsaba** [comment]
I"m looking into the file export issue.

**Comment by scsaba** [comment]
Well, I spent most of my day trying to find a solution to be able to export a file in Edge extension, but it seems it is not feasible at the moment. I tried several approach ([this is the preferred way](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/8178877/) in other browsers), but all of them failed with exception (either "script16386: no such interface supported" or "SEC7134: Resource not allowed to load."). So it seems saving a file (/blob) is not possible in an Edge extension. The solutions I tried, work under normal conditions (part of a homepage opened in Edge), so I'm not sure if it is intentional.

One option if you would like to support Edge is removing the possibility of downloading files (seed words, private key, state logs if I'm not mistaken) in Edge. It is far from ideal, but might be tolerable, as these are mostly convenience features.

The address copying is working, please let me know what issue you see with that (steps to reproduce).

**Comment by danfinlay** [comment]
Cool, once we confirm copying working, I think we can award this. Thanks a lot!

**Comment by scsaba** [comment]
Hi @tmashuang,

Please let me know what issue you had/have with address copying. I can copy the address using the context menu:

copyaddress

Thanks!

**Comment by scsaba** [comment]
@tmashuang meanwhile I checked it using the beta interface, and it also works there.
CC @danfinlay

**Comment by tmashuang** [comment]
The issue is not with the copy address functionality but with the seed phrase downloading and the state logs.
I will paste the screens.
The extension API that edge doesn't support is `runtime.getPlatformInfo()` which was the issue on my first initial run.
settingscreen
saveseedscreen

**Comment by scsaba** [comment]
@tmashuang Yes, runtime.getPlatformInfo() is not supported either, but as I noticed that does not break anything and does not prevent the user from doing anything. Unfortunately saving a file from an edge extension [is not possible currently](https://github.com/MetaMask/metamask-extension/pull/3309#issuecomment-369623726).

@danfinlay it seems there was some confusion [here](https://github.com/MetaMask/metamask-extension/pull/3309#pullrequestreview-100230628), and Thomas [confirmed](https://github.com/MetaMask/metamask-extension/pull/3309#issuecomment-369793433) that address copying is working.

Please let me know if anything more is needed from my side.

**Comment by danfinlay** [comment]
Looking great @scsaba.

If you could just add `browserify-unibabel` to the `package.json` dependencies, I think we can wrap this up and call it complete. I've contacted our friend at Microsoft about file downloading, so we'll deal with that on our end.

**Comment by scsaba** [comment]
@danfinlay Now it is added. Sorry I missed this. It was added indirectly by browser-passworder, so the build passed.

**Review by danfinlay — APPROVED** [review]
Looks good, thanks a lot!

**Pull request merged**